x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateDiffTests.java

```diff
@@ -35,76 +36,84 @@ public static Iterable<Object[]> parameters() {
-                        ),
-                        "DateDiffEvaluator[unit=Attribute[channel=0], startTimestamp=Attribute[channel=1], "
-                            + "endTimestamp=Attribute[channel=2]]",
-                        DataType.INTEGER,
-                        equalTo(88170)
-                    )
-                ),
-                new TestCaseSupplier(
-                    "Date Diff Error Type unit",
-                    List.of(DataType.INTEGER, DataType.DATETIME, DataType.DATETIME),
-                    () -> TestCaseSupplier.TestCase.typeError(
-                        List.of(
-                            new TestCaseSupplier.TypedData(new BytesRef("seconds"), DataType.INTEGER, "unit"),
-                            new TestCaseSupplier.TypedData(zdtStart.toInstant().toEpochMilli(), DataType.DATETIME, "startTimestamp"),
-                            new TestCaseSupplier.TypedData(zdtEnd.toInstant().toEpochMilli(), DataType.DATETIME, "endTimestamp")
-                        ),
-                        "first argument of [] must be [string], found value [unit] type [integer]"
-                    )
-                ),
-                new TestCaseSupplier(
-                    "Date Diff Error Type startTimestamp",
-                    List.of(DataType.TEXT, DataType.INTEGER, DataType.DATETIME),
-                    () -> TestCaseSupplier.TestCase.typeError(
-                        List.of(
-                            new TestCaseSupplier.TypedData(new BytesRef("minutes"), DataType.TEXT, "unit"),
-                            new TestCaseSupplier.TypedData(zdtStart.toInstant().toEpochMilli(), DataType.INTEGER, "startTimestamp"),
-                            new TestCaseSupplier.TypedData(zdtEnd.toInstant().toEpochMilli(), DataType.DATETIME, "endTimestamp")
-                        ),
-                        "second argument of [] must be [datetime], found value [startTimestamp] type [integer]"
-                    )
-                ),
-                new TestCaseSupplier(
-                    "Date Diff Error Type endTimestamp",
-                    List.of(DataType.TEXT, DataType.DATETIME, DataType.INTEGER),
-                    () -> TestCaseSupplier.TestCase.typeError(
-                        List.of(
-                            new TestCaseSupplier.TypedData(new BytesRef("minutes"), DataType.TEXT, "unit"),
-                            new TestCaseSupplier.TypedData(zdtStart.toInstant().toEpochMilli(), DataType.DATETIME, "startTimestamp"),
-                            new TestCaseSupplier.TypedData(zdtEnd.toInstant().toEpochMilli(), DataType.INTEGER, "endTimestamp")
-                        ),
-                        "third argument of [] must be [datetime], found value [endTimestamp] type [integer]"
-                    )
+        List<TestCaseSupplier> suppliers = new ArrayList<>();
+        suppliers.add(
```

Comment:
Looks like we have a `List.of()` hater here :hehe:
_(Feel free to ignore this comment)_

Reply:
Usually not! But in these I find it's more readable to do it this way....